### DISCUSSION TITLE
What is the difference between `add` and `force`?

### DISCUSSION THREAD

**Discussion by mid-kid** [uncrustify/uncrustify#4033] [Q&A] [answered]
I swear I've looked everywhere in /usr/share/doc, in the documentation/ folder, the github wiki, the readme and other bits of documentation I can find, and I can't find any explanation of what the 5-state options mean (`ignore/add/remove/force/not_defined`)

Can anyone explain?

**Comment by mid-kid** [comment, marked as answer]
The end of the default config explains this:
```
# Meaning of the settings:
#   Ignore - do not do any changes
#   Add    - makes sure there is 1 or more space/brace/newline/etc
#   Force  - makes sure there is exactly 1 space/brace/newline/etc,
#            behaves like Add in some contexts
#   Remove - removes space/brace/newline/etc
```

**Comment by guy-maurel** [comment]
Have a look to htdocs/configuration.txt
